docs/src/design-principles.md

```diff
@@ -0,0 +1,21 @@
+# Design Principles
+
+The library is built on several key principles:
+
+* **Radical simplicity**: Each data structure follows the most straightforward design that solves the problem at hand.
+This makes the code easier to understand, debug, and maintain.
+
+* **Backward compatibility**: Upgrading the library version must preserve the data.
+All data structures have a metadata section with the layout version, ensuring that new versions can read data written by old versions.
+
+* **No `pre_upgrade` hooks**: A bug in the `pre_upgrade` hook can make your canister non-upgradable.
+The best way to avoid this issue is not to have a `pre_upgrade` hook at all.
```

Comment:
Should there be a link to the interface spec where the `pre_upgrade` hook is explained as this is the first mention of it in this book?

Reply:
Agreed. Done.